study1/week1/merge-nodes-in-between-zeros.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Definition for singly-linked list.
+ * public class ListNode {
+ *     int val;
+ *     ListNode next;
+ *     ListNode() {}
+ *     ListNode(int val) { this.val = val; }
+ *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
+ * }
+ */
+class Solution {
+    public ListNode mergeNodes(ListNode head) {
+        //time complexity O(n^2) -> because of while loop inside while loop
+        //space omplexity O(n) -> because it creates n nodes..? (head has n nodes..)
```

Comment:
Basically what you did is an in-place merging of nodes instead of constructing new nodes.
This means that the space complexity is `O(1)`, when calculating space complexity, you do not count the space already allotted for inputs.